component-tools/src/main/java/org/talend/sdk/component/tools/ComponentValidator.java

```diff
@@ -1161,6 +1168,45 @@ private void validateActions(final AnnotationFinder finder, final Set<String> er
                         .collect(toSet()));
     }
 
+    private void validateExceptions(final Set<String> errors) {
+        System.out.println("Validate exceptions");
+
+        boolean exceptionFound = Arrays
+                .stream(classes)
+                .flatMap(f -> streamClassesInDirectory(null, f))
+                .filter(ComponentException.class::isAssignableFrom)
+                .findFirst()
+                .isPresent();
+
+        if (!exceptionFound) {
+            errors.add("Component should declare a custom ComponentException;");
```

Comment:
You need to provide at least in which class there's a problem for helping dev to correct. Otherwise, the message is not comprehensible... 

**TODO**
- [ ] We should also work on the wording, it's not expressive and clear enough!

Reply:
@rlecomte-talend , please change error message to : 
> Component should declare a custom Exception that inherits from ComponentException.

Reply:
Done